### PR TITLE
[release-1.7] overcome transient errors in ci-entrypoint.sh

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind flake

**What this PR does / why we need it**:

Manual cherry-pick of #3148.

This PR adds retries/idempotency into the ci-entrypoint.sh script that we maintain in order to build Kubernetes + Azure clusters on behalf of various k8s upstream projects.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: scripts/ci-entrypoint.sh [hunk 1]
```diff
@@ -169,60 +175,49 @@ install_addons() {
...
     fi
+    "${HELM}" upgrade calico --install --repo https://projectcalico.docs.tigera.io/charts tigera-operator -f "${CALICO_VALUES_FILE}" --set-string "${CIDR_STRING_VALUES}" --namespace calico-system
+}
 
-    "${HELM}" repo add projectcalico https://projectcalico.docs.tigera.io/charts
-    "${HELM}" install calico projectcalico/tigera-operator -f "${CALICO_VALUES_FILE}" --set-string "${CIDR_STRING_VALUES}" --namespace tigera-operator --create-namespace
-
-    # Add FeatureOverride for ChecksumOffloadBroken in FelixConfiguration.
-    # This is the recommended workaround for https://github.com/projectcalico/calico/issues/3145.
-    "${KUBECTL}" apply -f "${REPO_ROOT}"/templates/addons/calico/felix-override.yaml
```
Comment (CecileRobertMichon) [inline scripts/ci-entrypoint.sh:178]: I was going to comment that we didn't cherry-pick https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3115 but we could have so this is 👍

**Review by CecileRobertMichon — COMMENTED** [review]
was about to do this but you beat me to it! Thanks

/lgtm
/approve

**Comment by mboersma** [comment]
/retest

[config-map not found](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/3155/pull-cluster-api-provider-azure-e2e-v1beta1/1623314786974437376) error.

**Pull request merged**